### ISSUE TITLE
[clang-tidy][false-positive] `misc-const-correctness` incorrectly flags pointer passed to "pointer to pointer"

### ISSUE THREAD

**Issue by tearfur** [llvm/llvm-project#170033]
`misc-const-correctness` incorrectly flags pointer to non-const (e.g. `int*`) that are passed to pointer to pointer parameters (e.g. `int**`).

https://godbolt.org/z/8Mj4GGc5v:

```
<source>:7:5: warning: pointee of variable 'ip' of type 'int *' can be declared 'const' [misc-const-correctness]
    7 |     int* ip = &i;
      |     ^  
      |         const
1 warning generated.
```

Current trunk: b22825631293c19f70ca9969bd9de6094c688430

**Comment by vbvictor** [comment]
Related to https://github.com/llvm/llvm-project/issues/131599 but not exactly duplicate.
